tests/test_models.py

```diff
@@ -131,6 +131,22 @@ class TestDevice:
                 2,
                 True,
             ),
+            (
+                "eliot://ELIOT-000000000000000000000000000ABCDE/00000000000000000000000000125abc",
+                Protocol.ELIOT,
+                "ELIOT-000000000000000000000000000ABCDE",
+                "00000000000000000000000000125abc",
+                None,
+                False,
+            ),
+            (
+                "eliot://ELIOT-000000000000000000000000000ABCDE/00000000000000000000000000125abc#1",
+                Protocol.ELIOT,
+                "ELIOT-000000000000000000000000000ABCDE",
+                "00000000000000000000000000125abc",
+                1,
+                False,
```

Comment:
Can you please also add a test where the device url does not match?

Reply:
The idea is too avoid bad surprise if we retrieve a new kind of device url.

> I don't understand how I should do it
Just add a new set of parameters to your test

> how should I define which result to test against for a bad result
That’s you that defined the result when you implemented the logic in Device class :)
If any error is raised, it means we are doing something bad.

I guess, the following set of parameters should work:

```
(
                "foo://whatever-blah/12",
                None,
                "whatever-blah,
                "12",
                None,
                False,
            ),
```

```
(
                "foo://whatever",
                None,
                None,
                None,
                None,
                False,
            ),
```

Reply:
Done.
I have to update init of vars to comply with mypy and wrong test case